Add blurb to README.md clarifying that only Python versions >=3.6 are supported

jsf9k opened the pull request:
## 🗣 Description

Add a blurb to `README.md` stating that only Python versions 3.6 and greater are supported.

## 💭 Motivation and Context

This change is motivated by the discussion in #1.

## 🧪 Testing

I ran all the pre-commit hooks and verified that they all passed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add blurb clarifying that only Python versions >=3.6 are supported

dav3r reviewed (approved):
Sorry Python 2, your time has come and gone! Roll with the changes. 🐍🐍🐍

jsf9k committed:
Create a "Requirements" section instead of a one-line note

Thanks to @felddy for the suggestion.

dav3r reviewed (approved):
👍

felddy reviewed (dismissed):
Strong 2020 work!  🦾

felddy reviewed (changes requested):
Why don't we make this explicit in `setup.py` with a `python_requires='>=3.6'` 

https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires
https://stackoverflow.com/questions/45778790/specifying-the-minimum-python-version-required-for-a-module-in-pip

Probably backport this to the skeleton too.

jsf9k commented:
> Why don't we make this explicit in `setup.py` with a `python_requires='>=3.6'`
> 
> https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires
> https://stackoverflow.com/questions/45778790/specifying-the-minimum-python-version-required-for-a-module-in-pip
> 
> Probably backport this to the skeleton too.

@felddy, do you intend this to be in addition to the "Requirements" section or instead of it?

felddy committed:
Add a python_requires to prevent pip from installing on < 3.6
See: 
https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

felddy commented:
In addition.  I thought you were away.  I just pushed the change up in 
8c610d4
